f/radio/fn_radioChannels.sqf

```diff
@@ -0,0 +1,89 @@
+// F3 - Folk ARPS Long-Range Radio Module
+// Credits and documentation: https://github.com/folkarps/F3/wiki
+
+/* ========================
+This module is activated in init.sqf.
+Example:
+[true] spawn f_fnc_radioChannels;
+
+Arguments:
+0. Long-range split mode (boolean). If true, each radio backpack type has a separate channel. If false, all backpacks use one channel. Default false.
+1. Number of channels to create (number, optional). If this is set, the number of channels to be created will be limited to this number. If channels are not split, this will be automatically set to 1.
+
+This module requires some missionmaker configuration. If you want to enable vehicle radio access, you need to specify vehicle classes or variable names in "f\radio\f_channelsList.sqf".
+=========================== */
+
+params [["_splitMode",false],["_channelCount",10]];
+
+// Disable the Command channel to avoid cheating
+2 enableChannel false;
+
+if (isServer) then {
+
+	// Make the split mode something we can reference later
+	f_var_radioSplitMode = _splitMode;
+	publicVariable "f_var_radioSplitMode";
+	
+	// Make the channel count something we can reference later
+	f_var_radioChannelCount = _channelCount;
+	publicVariable "f_var_radioChannelCount";
+
+	// This will be used later
+	f_var_radioChannelUnified = [];
+	
+	// Define the list of channel properties. Go to this file to change which items and vehicles grant radio access.
+	#include "f_channelsList.sqf";
+	
+	f_var_radioChannels = createHashmap;
+	f_var_radioChannels set [1, [_channelName1,_channelColour1, (_channelList1 apply {toLower _x})]];
```

Comment:
The programmer in me wants to suggest making this a loop with `compile` but I'll resist :joy: